src/pyscaffold/shell.py

```diff
@@ -191,7 +214,39 @@ def get_editor(**kwargs):
 
 def edit(file: PathLike, *args, **kwargs) -> Path:
     """Open a text editor and returns back a :obj:`Path` to file, after user editing."""
-    editor = ShellCommand(get_editor())
+    editor = ShellCommand(_quote(get_editor()))
+    # ^  since we quote here we cannot pass flags in EDITOR environment variable
```

Comment:
I don't know how popular is to add flags to the `EDITOR` variable, but that is quite common with other variables.
For example, right now I have `PAGER=less -R` in my env.

This change prevents any attempt to add flags...

Reply:
I got a bad feeling with quoting here due to the implications you mentioned. One would trade the flexibility under Mac/Linux for fixing some Windows limitations.

Reply:
One thing important to mention is that the problem with the spaces in the executable path can also be observed in POSIX environments 😝 

As long as we use `subprocess.run` with `shell=True`, we do need to escape the whitespaces, and therefore we prevent the ability to set flags via env vars.

One way around that, would be quoting the result in the `get_editor`, only when we know it does not come from environment variables, more specifically here:
https://github.com/pyscaffold/pyscaffold/blob/1f7f64f94b567e8bb6728d4cc5ad847f84366df9/src/pyscaffold/shell.py#L182

```python
# instead we could do
others = (_quote(get_executable(e)) for e in EDITORS)
```
and then require the users to do proper quoting if they rely on environment variables for `EDITOR` or `VISUAL`.

However we are coupling `get_editor` with a specific way of calling `ShellCommand`... If we are only interested in the raw value of the editor executable path (for example to compare with other path) then we end up with extra problematic quotes (e.g. if later we decide to call `subprocess.run` or even instantiate `ShellCommand` with `shell=False` and an arg list, we would have an error because quotes are not valid chars in an executable path).

Reply:
Apparently, using flags in `EDITOR` is a thing:
- https://dev.to/jonasbn/til-integrate-visual-studio-code-with-shell--cli-2l1l
- https://docs.github.com/en/github/getting-started-with-github/associating-text-editors-with-git
- https://stackoverflow.com/questions/8951275/how-can-i-make-sublime-text-the-default-editor-for-git